R/mr.R

```diff
@@ -6,107 +6,110 @@ calc_I2 = function(Q, Est) {
-#' of susie result. "X_ID" and "gene_id" are ensemble ID and gene name respectively.
-#' 
-#' @importFrom qvalue qvalue
-#' @import dplyr
+#' Description of what the function does.
+#'
+#' @param susie_result A list containing the results of SuSiE analysis. This list should include nested elements such as 'susie_results', 'susie_result_trimmed', and 'top_loci', containing details about the statistical analysis of genetic variants.
+#' @param condition A character string specifying the conditions. This is used to select the corresponding subset of results within 'susie_result'.
+#' @param gwas_sumstats_db A data frame containing summary statistics from GWAS studies. It should include columns for variant id and their associated statistics such as beta coefficients and standard errors.
+#' @param sets A character string indicating the method used to define sets of genetic variants. Defaults to "sets". This parameter is used to specify the type of sets to extract from the 'susie_result' object.
+#' @param coverage A character string specifying the coverage threshold for credible sets, used when 'sets' is not "sets". Defaults to "coverage_0.95", indicating a 95% coverage credible set.
+#' @param allele_qc Optional. A logical value indicating whether allele qc should be performed on the variants. When TRUE, allele qc are applied to the variants based on the GWAS summary statistics database ('gwas_sumstats_db').
+#' @return A data frame formatted for MR analysis or NULL if cs_list is empty.
 #' @export
-twas_mr_format_input <- function(target_genes, susie_path, weights_file_path) {
-  mr_format_input <- NULL
-  for (k in 1:dim(target_genes)[1]) {
-    gene_name <- target_genes$gene_name[k]
-    gene_id <- target_genes$gene_id[k]
-    qtl_susie_res <- readRDS(susie_path$path[susie_path$ID == gene_name])
-    
-    if (!is.null(names(qtl_susie_res[[1]]$sets$cs))) {
-      qtl_finemap <- qtl_susie_res[[1]]$top_loci %>% 
-        filter(cs_index_primary >= 1) %>% 
-        select(-cs_index_secondary) %>% 
-        mutate(X_ID = gene_name) %>% 
-        rename("snp" = "variant_id", "bhat_x" = "bhat", "sbhat_x" = "sbhat", "cs" = "cs_index_primary")
-      
-      outcome_QC <- readRDS(weights_file_path$path[weights_file_path$ID == gene_name])$outcome_QC
-      merge_snp <- intersect(qtl_finemap$snp, outcome_QC$variant_allele_flip)
-      
-      format_input <- qtl_finemap[match(merge_snp, qtl_finemap$snp), ] %>% 
-        mutate(gene_id = gene_id) %>% 
-        cbind(., outcome_QC[match(merge_snp, outcome_QC$variant_allele_flip), ] %>% 
-                select(beta, se) %>% 
-                rename("bhat_y" = "beta", "sbhat_y" = "se"))
-      
-      mr_format_input <- rbind(mr_format_input, format_input)
+mr_format<- function(susie_result, condition, gwas_sumstats_db, sets = "sets", coverage = "coverage_0.95", allele_qc = TRUE) {
```

Comment:
Actually @AnqiWang2021 i am not sure why you need sets because the coverage table will contain everything including all variants inside of the 0.95 CS?